### PR TITLE
dhclient: support search domain options

### PR DESCRIPTION
**Pull request by isi-lincoln**
Add support for dhcp options 15 and 119. Adds the domain and search
options to resolv.conf.

Signed-off-by: Lincoln Thurlow <lincoln@isi.edu>

### CODE REVIEW

File: pkg/dhclient/dhclient.go [hunk 1]
```diff
@@ -108,18 +108,35 @@ func Configure6(iface netlink.Link, packet *dhcpv6.Message) error {
...
+		rc.WriteString(fmt.Sprintf("domain %s\n", domain))
+	}
+	if ns != nil {
+		for _, ip := range ns {
+			rc.WriteString(fmt.Sprintf("nameserver %s\n", ip))
+		}
+	}
+	if sl != nil {
+		rc.WriteString("search ")
+		for index, label := range sl {
```
Comment (hugelgupf) [inline pkg/dhclient/dhclient.go]: rc.WriteString(strings.Join(sl, " "))

instead of he whole loop

File: pkg/dhclient/dhcp4.go [hunk 2]
```diff
@@ -107,7 +123,9 @@ func (p *Packet4) Lease() *net.IPNet {
 }
 
 var (
-	ErrNoBootFile       = errors.New("no boot file name present in DHCP message")
+	//ErrNoBootFile represents that no pxe boot file was found
```
Comment (hugelgupf) [inline pkg/dhclient/dhcp4.go]: nit: space it and end with punctuation

```go
var (
  // ErrNoBootFile represents that no PXE boot file was found.
  ErrNoBootFile =

  // ErrNoServerHostName ...
  ErrNoServerHostName = ...
```

(did gofmt not do the spacing on its own?)

File: pkg/dhclient/dhclient.go (hunk 1 above)
Comment (isi-lincoln) [reply pkg/dhclient/dhclient.go]: sounds good.

File: pkg/dhclient/dhcp4.go (hunk 2 above)
Comment (isi-lincoln) [reply pkg/dhclient/dhcp4.go]: Full disclosure, I use `vim-go` which does my formatting (including gofmt), it did not add spaces.  I will do so now.

**Comment by isi-lincoln** [comment]
High level comment.  I left ipv6 as-is with respect to dns domain and search list as I did not test it.  That may be a todo for later, I suspect it too can plug in with GatherDNSSettings depending on `dhcp` library for ipv6 packets.

**Comment by hugelgupf** [comment]
Re v6 I figured but nothing here precludes it from being implemented at some point.

**Pull request merged**